doc/source/make_tables.py

```diff
@@ -1488,16 +1488,9 @@ def fetch_multiblock(cls, kind: Literal['hetero', 'homo', 'single']):
             if pv.MultiBlock in types_list:
                 types_list.remove(pv.MultiBlock)
                 num_datasets = len(types_list)
-                if (
-                    num_datasets == 1
-                    and kind == 'single'
-                    or (
-                        num_datasets >= 2
-                        and len(set(types_list)) == 1
-                        and kind == 'homo'
-                        or len(set(types_list)) > 1
-                        and kind == 'hetero'
-                    )
+                if (num_datasets == 1 and kind == 'single') or (
+                    (num_datasets >= 2 and len(set(types_list)) == 1 and kind == 'homo')
+                    or (len(set(types_list)) > 1 and kind == 'hetero')
                 ):
```

Comment:
Make the logic easier to follow. I ran this locally and checked that the generated tables add up correctly in the dataset gallery (i.e. n_homo + n_hetero + n_single == n_multiblock)
```suggestion
                num_types = len(set(types_list))

                is_single = num_datasets == 1
                is_homo = num_datasets >= 2 and num_types == 1
                is_hetero = num_datasets >= 2 and num_types > 1
                if (is_single and kind == 'single') or (is_homo and kind == 'homo') or (is_hetero and kind == 'hetero'):
```